src/main/java/xyz/nucleoid/stimuli/mixin/block/DispenserBlockMixin.java

```diff
@@ -0,0 +1,41 @@
+package xyz.nucleoid.stimuli.mixin.block;
+
+import net.minecraft.block.DispenserBlock;
+import net.minecraft.block.entity.DispenserBlockEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPointerImpl;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+import xyz.nucleoid.stimuli.Stimuli;
+import xyz.nucleoid.stimuli.event.block.DispenserUseEvent;
+
+@Mixin(DispenserBlock.class)
+public class DispenserBlockMixin {
+
+    @Inject(
+            method = "dispense",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/block/DispenserBlock;getBehaviorForItem(Lnet/minecraft/item/ItemStack;)Lnet/minecraft/block/dispenser/DispenserBehavior;",
+                    shift = At.Shift.BEFORE
+            ),
+            locals = LocalCapture.CAPTURE_FAILSOFT,
```

Comment:
Mixins for events should fail early as events should never be skipped.

```suggestion
            locals = LocalCapture.CAPTURE_FAILHARD,
```